pageserver/src/walingest.rs

```diff
@@ -424,8 +457,16 @@ impl WalIngest {
                 let info = decoded.xl_info & pg_constants::XLR_RMGR_INFO_MASK;
                 if info == pg_constants::XLOG_RUNNING_XACTS {
                     let xlrec = crate::walrecord::XlRunningXacts::decode(&mut buf);
-                    self.checkpoint.oldestActiveXid = xlrec.oldest_running_xid;
-                    self.checkpoint_modified = true;
+                    let modified =
+                        enum_pgversion_dispatch!(&mut self.checkpoint, CheckpointVersion, cp, {
+                            let modified = cp.oldestActiveXid != xlrec.oldest_running_xid;
+                            cp.oldestActiveXid = xlrec.oldest_running_xid;
+                            modified
+                        });
+
+                    if modified {
+                        self.checkpoint_modified = true;
+                    }
```

Comment:
This is a drive-by change that makes the `checkpoint_modified` updating more precise than it was before.
Not wrong, just pointing it out and seeking confirmation.